akka-cluster/src/main/scala/akka/cluster/ClusterRemoteWatcher.scala

```diff
@@ -111,13 +118,16 @@ private[cluster] class ClusterRemoteWatcher(
     if (m.address != selfAddress) {
       quarantineOldIncarnation(m)
       clusterNodes += m.address
+      if (tombstones.contains(m.address)) // new node with same address joined
+        tombstones -= m.address
       takeOverResponsibility(m.address)
       unreachable -= m.address
     }
 
   def memberRemoved(m: Member, previousStatus: MemberStatus): Unit =
     if (m.address != selfAddress) {
       clusterNodes -= m.address
+      tombstones += (m.address -> System.currentTimeMillis())
```

Comment:
`Gossip.tombstones` also contains similar, but it would be difficult to access that from here.

I see one problem with tracking the tombstones here. Example:
- node1, node2, node3
- node3 is removed
- node4 is joining
- from node4 a watch is performed for an ActorRef that was previously on node3
- node4 doesn't know that node3 was part of the cluster previously, because there will be no MemberRemoved in node4 since it joined after the removal

Therefore it would be better to use the `Gossip.tombstones`, because that would also be propagated to node4. Maybe we can set it in `ClusterReadView`, where we already have some volatile access?

Reply:
I did figure out how to get the initial tombstones into `ClusterState`, but I don't know how it would be possible to keep it up to date, that is done through cluster event subscriptions and no sub gives access to the latest gossip?

Reply:
To follow the same pattern as the other things we would add another internal event TombstoneChanged for this purpose, or we would add the tombstones to the already internal event SeenChanged

Reply:
Given that the order of the events are important it might be best with a new TombstoneChanged event that the ClusterRemoteWatcher can subscribe to. `TombstoneChanged` can carry the full `Set[UniqueAddress]` in each event so that we don't have to care about diffs.

Reply:
Alright, I'll give that a shot. Thx for the hints

Reply:
Seems that worked out fine